src/main/resources/hudson/plugins/promoted_builds/parameters/PromotedBuildParameterDefinition/index.jelly

```diff
@@ -31,9 +31,9 @@ THE SOFTWARE.
 			<input type="hidden" name="name" value="${it.name}" />
             <select name="runId">
               <j:forEach var="run" items="${it.builds}">
-                <option value="${run.externalizableId}">${run}</option>
+                <option value="${run.externalizableId}">${run.fullDisplayName}</option>
```

Comment:
Do you really want `fullDisplayName`, which will include all parent folders/jobs ("Folder » SubFolder » Job $BuildDisplayName"), or just `displayName`, which would be either "Job #NUM" or the build's modified display name in your case?

Reply:
I went with `fullDisplayName` because jenkins parameters use that. In fact I need modified display name :)

Reply:
Ok, I will look around to see which seems more consistent across Jenkins, but other than that this PR seems totally fine to me, so I will try to merge+release it in the next week or so. Thanks!

Reply:
Glad to contribute :)